### ISSUE TITLE
Authorization checks are re-enabled when exceptions are caught

### ISSUE THREAD

**Issue by bwaidelich** [neos/flow-development-collection#2521]
### Description

Especially in CLI mode authorization checks should be disabled.
When code is wrapped in a `Security\Context::withoutAuthorizationChecks()` call and exceptions are caught, auth checks are no longer disabled in the outer closure leading to exceptions like
```
The security Context cannot be initialized yet
```

### Steps to Reproduce

```php
// this is the outer closure (e.g. triggered by CommandRequestHandler::handleRequest())
$this->securityContext->withoutAuthorizationChecks(function () {

  // some try/catch that swallows the exception, somewhere in the stack trace
  try {

    // some user code that should skip auth checks
    $this->securityContext->withoutAuthorizationChecks(function () {
      // ....and might lead to an exception
      throw new \Exception('Some exception');
    });
  } catch (\Exception $exception) {
  }

  // the bug (this should not be the case):
  $this->securityContext->areAuthorizationChecksDisabled() === false;
});
```

### Affected Versions

Flow: all (lowest supported branch: 6.3)

### Note:

Another reason not to silently swallow exceptions

**Comment by sorenmalling** [comment]
@bwaidelich Not automatically closed after merge of #2522  - can I close this, or does it need anything more?

**Comment by bwaidelich** [comment]
@sorenmalling yes, thanks for the heads up, I'll close the issue
